plugins/net_plugin/net_plugin.cpp

```diff
@@ -2182,14 +2182,27 @@ namespace eosio {
               sync_next_expected_num < sync_last_requested_num );
    }
 
+   // called from c's connection strand
    bool sync_manager::is_sync_request_ahead_allowed(block_num_type blk_num) const REQUIRES(sync_mtx) {
       if (blk_num >= sync_last_requested_num) {
          // do not allow to get too far ahead (sync_fetch_span) of chain head
          // use chain head instead of fork head so we do not get too far ahead of applied blocks
          uint32_t head = my_impl->get_chain_head_num();
          if (blk_num < head + sync_fetch_span)
             return true;
+
+         // might be in irreversible mode
+         controller& cc = my_impl->chain_plug->chain();
+         auto calculated_lib = cc.fork_db_head().irreversible_blocknum();
+         if (calculated_lib <= my_impl->get_chain_lib_num()) {
+            fc_ilog(logger, "sync ahead allowed past sync-fetch-span ${sp} for paused LIB ${l}, forkdb size ${s}",
+                    ("sp", sync_fetch_span)("l", calculated_lib)("s", cc.fork_db_size()));
+            return true;
+         }
```

Comment:
You can also dlog here why sync ahead not allowed.

Reply:
Done